### ISSUE TITLE
`for element` block has rendering issues.

### ISSUE THREAD

**Issue by CleoQc** [Microsoft/pxt-microbit#1749]
**Describe the bug**
Looking at step 4 of https://makecode.microbit.org/projects/crashy-bird, there's this block:
![mycode](https://user-images.githubusercontent.com/4965628/50177259-84bd6000-02cf-11e9-891e-09b030697699.png)

The green "obstacle" pulldown menu is nowhere to be found but there is an orange variable block called "obstacle". 

If I switch the step 4 code into javascript, and copy/paste the javascript code into my project, I get the green pulldown menu as shown in image above.
```let obstacles: game.LedSprite[] = []

basic.forever(() => {
    for (let obstacle of obstacles) {
        obstacle.change(LedSpriteProperty.X, -1)
    }
    basic.pause(1000)
})
```

If I fill the blocks "by hand" by dragging and dropping blocks, I get the bottom forever script: (top script left in for comparison purposes)
![obstacles](https://user-images.githubusercontent.com/4965628/50177644-820f3a80-02d0-11e9-8bd0-bb7e41d07204.png)

Yet the javascript is the same for both: (note the pause is not the same, to tell them apart)
```
basic.forever(function () {
    for (let obstacle of obstacles) {
        obstacle.change(LedSpriteProperty.X, -1)
    }
    basic.pause(1000)
})
basic.forever(function () {
    for (let obstacle of obstacles) {
        obstacle.change(LedSpriteProperty.X, -1)
    }
    basic.pause(2000)
})
```

To summarize, if the `for element` block is first written in javascript and then switched to blocks, it will not be generated in a way that can be duplicated by using blocks only. This leads to confusing explanations in tutorials.
